typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -128,31 +137,89 @@ export abstract class HyperlaneDeployer<
 
     this.logger(`Deploy ${contractName} on ${chain}`);
 
-    const contract = await factory
-      .connect(signer)
-      .deploy(...args, chainConnection.overrides);
+    if (
+      deployOpts &&
+      deployOpts.create2Salt &&
+      (await chainConnection.provider.getCode(CREATE2FACTORY_ADDRESS)) != '0x'
```

Comment:
should this throw?

Reply:
Why should this throw? In the `test` environment, the factory won't exist, so in that case, it should just deploy "normally"